python_modules/dagster/dagster/_core/ext/subprocess.py

```diff
@@ -54,15 +45,19 @@ def run(
         env: Optional[Mapping[str, str]] = None,
         cwd: Optional[str] = None,
     ) -> None:
-        external_context = ExtOrchestrationContext(context=context, extras=extras)
-        with self._setup_io(external_context, context_injector, message_reader) as io_env:
+        with ext_protocol(
+            context=context,
+            context_injector=context_injector or ExtTempFileContextInjector(),
+            message_reader=message_reader or ExtTempFileMessageReader(),
+            extras=extras,
+        ) as ext_context:
             process = Popen(
                 command,
                 cwd=cwd or self.cwd,
                 env={
-                    **self.get_base_env(),
-                    **{**(env or {}), **(self.env or {})},
-                    **io_env,
+                    **ext_context.get_external_process_env_vars(),
+                    **(self.env or {}),  # type: ignore  # (pyright bug)
```

Comment:
explain pyright bug?

Reply:
I don't know why pyright is complaining here but I know the issue is gone on higher pyright versions. So these comments will go away as soon as this lands: https://github.com/dagster-io/dagster/pull/16360

I'm just waiting to land that pyright update until the next ruff release (prob in a day or so) so I can update both simultaneously.

Reply:
Lovely that sounds great